### ISSUE TITLE
network: No static address is stored

### ISSUE THREAD

**Issue by zhangyoufu** [systemd/systemd#17304]
**systemd version the issue has been seen with**
 > 246.6-1-arch

**Used distribution**
 > Arch Linux

**Unexpected behaviour you saw**
 > dummy0: No static address is stored.
 > dummy0: Failed

**Steps to reproduce the problem**
1. prepare configuration files
   ```
   # /etc/systemd/network/dummy0.netdev
   [NetDev]
   Name=dummy0
   Kind=dummy
   ```
   ```
   # /etc/systemd/network/dummy0.network
   [Match]
   Name=dummy0

   [Address]
   Address=100.64.0.1/32
   Peer=100.64.0.2/32
   ```
2. `networkctl reload`. The `dummy0` interface is shown as `routable` and `configured` after IPv6 DAD.
3. `systemctl restart systemd-networkd`. The `dummy0` interface is shown as `routable` and `failed`.

This issue is somehow related to prefix length, and only affect point-to-point address.

**Comment by yuwata** [comment]
Could you provide debugging logs for the issue?

**Comment by yuwata** [comment]
(Initially I thought this is a duplicate of #16696, but it should be fixed by #16697 and its commits are backported in v246.3.)

**Comment by yuwata** [comment]
Ugh. Indeed. Now I can reproduce the error...

**Comment by yuwata** [comment]
```
Oct 12 08:34:06 systemd-networkd[4937]: dummy0: Setting address genmode done.
Oct 12 08:34:06 systemd-networkd[4937]: dummy0: Removing address 100.64.0.1
Oct 12 08:34:06 systemd-networkd[4937]: dummy0: Discovering IPv6 routers
Oct 12 08:34:06 systemd-networkd[4937]: dummy0: State changed: initialized -> configuring
Oct 12 08:34:06 systemd-networkd[4937]: dummy0: link_check_ready(): static addresses are not configured.
Oct 12 08:34:06 systemd-networkd[4937]: dummy0: link_check_ready(): static addresses are not configured.
Oct 12 08:34:06 systemd-networkd[4937]: dummy0: Configuring address: 100.64.0.1
Oct 12 08:34:06 systemd-networkd[4937]: dummy0: Setting addresses
Oct 12 08:34:06 systemd-networkd[4937]: dummy0: Forgetting address: 100.64.0.1/32 (valid forever)
Oct 12 08:34:06 systemd-networkd[4937]: dummy0: Forgetting route: dst: 100.64.0.2/32, src: n/a, gw: n/a, prefsrc: 100.64.0.1, scope: link, table: main, proto: kernel, type: unicast
Oct 12 08:34:06 systemd-networkd[4937]: dummy0: Forgetting route: dst: 100.64.0.1/32, src: n/a, gw: n/a, prefsrc: 100.64.0.1, scope: host, table: local, proto: kernel, type: local
Oct 12 08:34:06 systemd-networkd[4937]: dummy0: Remembering foreign address: 100.64.0.1/32 (valid forever)
Oct 12 08:34:06 systemd-networkd[4937]: dummy0: link_check_ready(): static addresses are not configured.
Oct 12 08:34:06 systemd-networkd[4937]: dummy0: Remembering route: dst: 100.64.0.1/32, src: n/a, gw: n/a, prefsrc: 100.64.0.1, scope: host, table: local, proto: kernel, type: local
Oct 12 08:34:06 systemd-networkd[4937]: dummy0: Remembering route: dst: 100.64.0.2/32, src: n/a, gw: n/a, prefsrc: 100.64.0.1, scope: link, table: main, proto: kernel, type: unicast
Oct 12 08:34:06 systemd-networkd[4937]: dummy0: Addresses set
Oct 12 08:34:06 systemd-networkd[4937]: dummy0: No static address is stored.
Oct 12 08:34:06 systemd-networkd[4937]: dummy0: Failed
Oct 12 08:34:06 systemd-networkd[4937]: dummy0: State changed: configuring -> failed
```

**Comment by yuwata** [comment]
Fix is waiting in #17316. But, unfortunately, the fix cannot be simply cherry-picked into v246-stable repository...
